packages/babel-plugin-transform-es2015-arrow-functions/src/index.js

```diff
@@ -10,7 +10,7 @@ export default function ({ types: t }) {
           node.type = "FunctionExpression";
 
           let boundThis = t.thisExpression();
-          boundThis._forceShadow = path;
+          boundThis._forceShadow = node;
```

Comment:
I'm a little worried this is a breaking g change since someone could end up with this in Babel core but an old version of other stuff that expected a NodePath, or the other way around.